parsing/g_prim.mlg

```diff
@@ -48,10 +48,16 @@ GRAMMAR EXTEND Gram
     bignat bigint natural integer identref name ident hyp preident
     fullyqualid qualid reference dirpath ne_lstring
     ne_string string lstring pattern_ident by_notation
-    smart_global bar_cbrace strategy_level fields;
+    smart_global bar_cbrace strategy_level fields scope_name scope_key;
   preident:
     [ [ s = IDENT -> { s } ] ]
   ;
+  scope_name:
+    [ [ s = IDENT -> { s } ] ]
```

Comment:
I don't see how aliasing a plain OCaml value to `IDENT` would show up in the grammar extracted by `doc_grammar`--and I don't know exactly what code you have in mind, either.

I don't see a factorization issue, either, other than that the grammar seems inconsistent about when it accepts `scope_name` and `scope_key` - `Open Scope` and `argument_spec` take either, but `term_scope` only takes `scope_key`.  I expect that tends to confuse users.  There are other examples.  Maybe it's worth making them more interchangeable outside of this PR.  (They may be inconsistent because the different handling of the `IDENT` is tedious to discover in the code.  This grammar change makes it much easier to see.)

Of course, when parsing `scope`, the parser will always recognize the `scope_name` production and never `scope_key`.

From `orderedGrammar`:

```
command: [
| "Open" "Scope" scope
  :
]

argument_spec: [
| OPT "!" name LIST0 ( "%" scope )
]

term_scope: [
| term0 "%" scope_key
]


scope: [
| scope_name
| scope_key
]
```